src/components/TodoFilter/TodoFilter.tsx

```diff
@@ -1,14 +1,22 @@
-import React from 'react';
+import { useAppDispatch, useAppSelector } from '../../app/hooks';
+import { actions as filterActions } from '../../features/filter';
+import { Status } from '../../types/Status';
 
 export const TodoFilter: React.FC = () => {
+  const dispatch = useAppDispatch();
+  const filter = useAppSelector(state => state.filter);
+
   return (
-    <form
-      className="field has-addons"
-      onSubmit={event => event.preventDefault()}
-    >
+    <form className="field has-addons">
       <p className="control">
         <span className="select">
-          <select data-cy="statusSelect">
+          <select
+            data-cy="statusSelect"
+            value={filter.status}
+            onChange={e => dispatch(
+              filterActions.setStatus(e.target.value as Status),
+            )}
```

Comment:
move this to the separate function
```suggestion
            onChange={setFilter}
```